pygmt/encodings.py

```diff
@@ -0,0 +1,124 @@
+charset: dict = {}
+
+# Adobe ISOLatin1+ charset.
+# Most characters are the same in ISOLatin1+ and ISO-8859-1 encodings.
+charset["ISOLatin1+"] = {
+    i: chr(i) for i in [*range(0o040, 0o177), *range(0o240, 0o400)]
+}
+# Handle characters that are different in ISOLatin1+ and ISO-8859-1 encodings.
+charset["ISOLatin1+"].update(
+    {
+        0o047: "\u2019",  # Change "Apostrophe" to "Right Single Quotation Mark"
+        0o055: "\u2212",  # Change "Hyphen-minus" to "Minus Sign"
+        0o140: "\u2018",  # Change "Grave Accent" to "Left Single Quotation Mark"
+        0o177: "\u0161",  # Set to "Latin Small Letter S with Caron"
+    }
+)
+# Add extended characters in ISOLatin1+.
+charset["ISOLatin1+"].update(
+    dict(
+        zip(
+            [*range(0o030, 0o040), *range(0o200, 0o240)],
+            "\ufffd\u2022\u2026\u2122\u2014\u2013\ufb01\u017e"
+            "\u0152\u2020\u2021\u0141\u2044\u2039\u0160\u203a"
+            "\u0153\u0178\u017d\u0142\u2030\u201e\u201c\u201d"
+            "\u0131\u0060\u00b4\u02c6\u02dc\u00af\u02d8\u02d9"
+            "\u00a8\u201a\u02da\u00b8\u0027\u02dd\u02db\u02c7",
+            strict=False,
+        )
+    )
+)
+
+# Adobe Symbol charset.
+charset["Symbol"] = dict(
+    zip(
+        [*range(0o040, 0o200), *range(0o240, 0o400)],
+        "\u0020\u0021\u2200\u0023\u2203\u0025\u0026\u220b"
+        "\u0028\u0029\u2217\u002b\u002c\u2212\u002e\u002f"
+        "\u0030\u0031\u0032\u0033\u0034\u0035\u0036\u0037"
+        "\u0038\u0039\u003a\u003b\u003c\u003d\u003e\u003f"
+        "\u2245\u0391\u0392\u03a7\u2206\u0395\u03a6\u0393"
+        "\u0397\u0399\u03d1\u039a\u039b\u039c\u039d\u039f"
+        "\u03a0\u0398\u03a1\u03a3\u03a4\u03a5\u03c2\u2126"
+        "\u039e\u03a8\u0396\u005b\u2234\u005d\u22a5\u005f"
+        "\uf8e5\u03b1\u03b2\u03c7\u03b4\u03b5\u03c6\u03b3"
+        "\u03b7\u03b9\u03d5\u03ba\u03bb\u03bc\u03bd\u03bf"
+        "\u03c0\u03b8\u03c1\u03c3\u03c4\u03c5\u03d6\u03c9"
+        "\u03be\u03c8\u03b6\u007b\u007c\u007d\u223c\ufffd"
+        "\u20ac\u03d2\u2032\u2264\u2215\u221e\u0192\u2663"
+        "\u2666\u2665\u2660\u2194\u2190\u2191\u2192\u2193"
+        "\u00b0\u00b1\u2033\u2265\u00d7\u221d\u2202\u2022"
+        "\u00f7\u2260\u2261\u2248\u2026\u23d0\u23af\u21b5"
+        "\u2135\u2111\u211c\u2118\u2297\u2295\u2205\u2229"
+        "\u222a\u2283\u2287\u2284\u2282\u2286\u2208\u2209"
+        "\u2220\u2207\u00ae\u00a9\u2122\u220f\u221a\u22c5"
+        "\u00ac\u2227\u2228\u21d4\u21d0\u21d1\u21d2\u21d3"
+        "\u25ca\u2329\u00ae\u00a9\u2122\u2211\u239b\u239c"
+        "\u239d\u23a1\u23a2\u23a3\u23a7\u23a8\u23a9\u23aa"
+        "\ufffd\u232a\u222b\u2320\u23ae\u2321\u239e\u239f"
+        "\u23a0\u23a4\u23a5\u23a6\u23ab\u23ac\u23ad\ufffd",
+        strict=False,
+    )
+)
+
+# Adobe ZapfDingbats charset.
+charset["ZapfDingbats"] = dict(
+    zip(
+        [*range(0o040, 0o220), *range(0o240, 0o400)],
+        "\u0020\u2701\u2702\u2703\u2704\u260e\u2706\u2707"
```

Comment:
In previous version, we maintained a big dictionary like:
```
 ✁✂✃✄☎✆✇✈✉☛☞✌✍✎✏
```
but here I decide to use:
```
\u0020\u2701\u2702\u2703\u2704\u260e\u2706\u2707
```
The main reasons are:

- Avoid having too many non-ASCII characters in the source code
- Some non-ASCII characters can't be displayed correctly in text editors or GitHub web UI